core/src/toga/style/applicator.py

```diff
@@ -17,14 +17,15 @@ def set_bounds(self):
             self.widget.layout.content_height,
         )
         for child in self.widget.children:
-            if child.applicator:
-                child.applicator.set_bounds()
+            child.applicator.set_bounds()
 
     def set_text_alignment(self, alignment):
         self.widget._impl.set_alignment(alignment)
 
     def set_hidden(self, hidden):
         self.widget._impl.set_hidden(hidden)
+        for child in self.widget.children:
+            child.applicator.set_hidden(hidden)
```

Comment:
Agreed a core test is worthwhile.

As for the behavior of a hidden child - I did think about that, and considered it "feature, not bug". As far as the HTML comparison is valid here, it would be consistent with HTML: If you hide a parent, the children aren't visible; but if you reparent a child, it becomes visible unless that child is *explicitly* hidden. It's probably worth documenting this... but I'm not sure where the right place would be - I guess it's a detail of Pack?

Reply:
FWIW: I've added a note about this to the Pack documentation on `visibility`.

Reply:
I wasn't talking about reparenting, I was talking about the case where the parent and the child are both *explicitly* hidden, and then the parent is made visible. It looks like this code will make the child visible as well.

Reply:
Ah - I missed the obvious counter case. That's going to be harder to handle. I'll see what I can come up with.

Reply:
In order to keep the applicator independent of the specific Pack implementation, the style object needs to provide an interpretation of "is hidden"; this should probably be pushed up to Travertino as an abstract method, but for now, Pack is the only viable implementation of a style language, so the matter is moot. However, I've been able to ensure that an explicitly hidden child isn't made visible because its parent is made visible. Docs have been updated to clarify, and there's a core and testbed test of this behavior.